### ISSUE TITLE
NPC reading issues

### ISSUE THREAD

**Issue by Funguss** [CleverRaven/Cataclysm-DDA#35532]
**Is your feature request related to a problem? Please describe.**  

Three main issues exist when asking an NPC to read.
1. The list of books are coloured based on your skills not theirs.
2. The pickup distance is one block from them and doesn't include books in your possession.
3. An NPC will read until they level up once instead of going until they're interrupted.

**Describe the solution you'd like**  

Sure, point 3 is how it is for the player, but it's significantly more annoying to set them reading again due to point one, and they'll probably be interrupting you while you're doing something else. Point 2 is also how it is for the player, but again this is simply frustrating and increases micro management. I see no valid reason not to extend the radius and have them walk to it or abstract it away in a similar fashion to how recipes work. Point one, I expect, would be more complicated to implement but would have the higher QOL effect.

**Describe alternatives you've considered**  

Granted I haven't made much headway with faction camps, but a library / school system would be great. Or just a dialogue option to get NPCs to read from the book stockpile of whatever will raise their skills. Either of these would address all the previous points to some extent, though anything to do with faction camps requires a significant time investment first.

**Comment by jasonmbrown** [comment]
I like the thought of a library/school, Maybe start it as a "training" building that later upgrades into a school. (Quite a few thoughts on a library/school building actually).
I can agree though that it can be tedious teaching npcs.

**Comment by sdp0et** [comment]
I wasn't even aware they had a pickup distance.  I've been manually trading to give them the books, but agree it would be nicer to treat them like tools for other tasks.

**Comment by LeahLuong** [comment]
Yeah, I came across this set of issues still in 0.F-3 & was about to post about them but decided to search 1st. The 1st issue is especially annoying because the list does see NPC skill lvls but colors the list according to your char's stats. The list is ordered according to this color scheme so if there are books you can't read yet but the NPC can, then they are shoved to the bottom.

I'd like to see the NPC reading system revamped in such a way that you merely have to select the skill you'd like them to read about & they choose the book & go to it until interrupted. Of course, skills they can't advance through your current library wouldn't be listed for selection.

**Comment by Brambor** [comment]
> I'd like to see the NPC reading system revamped in such a way that you merely have to select the skill you'd like them to read about & they choose the book & go to it until interrupted. Of course, skills they can't advance through your current library wouldn't be listed for selection.

I like this. Make a zone "books for reading". Make a menu "learn by reading" that uses these books. The usage is selecting what skills (not only one skill) to train to what level. Maybe select order of learning to:

1. Cooking to 3
2. Fabrication to 5
3. Cooking to 5

On selecting the skills, the menu will determine whether sufficient books are available: E.g. learn cooking from 0 to 2, from 2 to 3, from 3 to 5.

Based on NPC intelligence and book order, estimate time needed for NPC to learn all the stuff.

If a book goes missing, try using another book (maybe not as good) before failing. In that case, write to the log "X could find book Y so they are reading Z instead."

The menu "learn by reading" should be usable for NPCs and the player character. It can be useful to estimate time needed to gain skills by reading.

**Comment by Brambor** [comment]
To make NPC read a book:
1. `C` talk to someone
2. `t`alk to
3. Select NPC
4. `d` There's something I want you to do.
5. `l` Let's talk about your current activity.
6. `g` Please study from a book.
   - There are two more options:
   ![Image](https://github.com/user-attachments/assets/ffa42112-3b74-4e60-9474-3bc043e5ee6e)

**Comment by Brambor** [comment]
> The list of books are coloured based on your skills not theirs.

No longer true.

![Image](https://github.com/user-attachments/assets/52c4fe49-462e-42f0-9dad-577be258ca96)

- OS: Windows
    - OS Version: 10.0.19045.5371 (22H2)
- Game Version: 0.G-16437-g458b9fd224-dirty [64-bit] - master was 8fe32dee74
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth]
]
